### PR TITLE
Move Metadata Storage to Channel Data

### PR DESCRIPTION
**Pull request by ncteisen**
Store metadata on channel since it does not change per call

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                              FILE SIZE
 ++++++++++++++ GROWING                                                ++++++++++++++

 -------------- SHRINKING                                              --------------
  -2.7%    -131 src/core/ext/filters/http/client/http_client_filter.cc    -131  -2.7%
      -7.5%    -129 hc_start_transport_stream_op_batch                        -129  -7.5%
      -8.8%     -12 init_call_elem                                             -12  -8.8%
      -3.4%      -3 [Unmapped]                                                  -3  -3.4%
      -3.1%      -3 recv_initial_metadata_ready                                 -3  -3.1%
      -3.1%      -3 recv_trailing_metadata_ready                                -3  -3.1%

 -+-+-+-+-+-+-+ MIXED                                                  +-+-+-+-+-+-+-
  +0.0%      +3 [None]                                                    -781  -0.0%

  -0.0%    -128 TOTAL                                                     -912  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,980,579      Total (<)      1,980,683

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,811,770      Total (>)     10,811,761

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] Performance differences noted:
Benchmark                                                               cpu_time    real_time
----------------------------------------------------------------------  ----------  -----------
BM_PumpStreamClientToServer<InProcess>/262144                           +4%         +4%
BM_PumpStreamServerToClient<InProcess>/262144                           +4%         +4%
BM_StreamingPingPongMsgs<MinInProcess, NoOpMutator, NoOpMutator>/32768  +6%         +6%
```

File: src/core/ext/filters/http/client/http_client_filter.cc [hunk 1]
```diff
@@ -364,43 +356,28 @@ static void hc_start_transport_stream_op_batch(
...
        layer headers. */
     error = grpc_metadata_batch_add_head(
         batch->payload->send_initial_metadata.send_initial_metadata,
-        &calld->method, method);
+        &calld->method_storage, method);
     if (error != GRPC_ERROR_NONE) goto done;
     error = grpc_metadata_batch_add_head(
         batch->payload->send_initial_metadata.send_initial_metadata,
-        &calld->scheme, channeld->static_scheme);
+        &channeld->scheme_storage, channeld->static_scheme);
```
Comment (hcaseyal) [inline src/core/ext/filters/http/client/http_client_filter.cc]: I'm a little confused on what you're trying to do here. You're moving the metadata storage to the channel data, but setting it to the metadata elem in hc_start_transport_stream_op_batch, which is called per call.

**Pull request closed without merging**